### PR TITLE
AC-909 Converted AddEditPatient Package To Kotlin

### PR DESCRIPTION
**Pull request by Rohit-2602**
## Issue I worked on

JIRA Issue: https://issues.openmrs.org/browse/AC-909

## Checklist: I completed these to help reviewers :)

- [ ] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

- [ ] I have **added tests** to cover my changes. (If you refactored
existing code that was well tested you do not have to add tests)

- [x] All new and existing **tests passed**.

- [x] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientActivity.kt [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        }
+
+        //Check if bundle includes patient ID
+        var patientBundle = savedInstanceState
+        if (patientBundle != null) {
+            patientBundle.getString(ApplicationConstants.BundleKeys.PATIENT_ID_BUNDLE)
+        } else {
+            patientBundle = intent.extras
+        }
+        var patientID: String = "PatientID"
```
[the remark this answers is not in the document]
Comment (SalahoAmro) [reply openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientActivity.kt:60]: Also, the original code does not assign a value to the variable more than an empty String.

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientActivity.kt [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        if (patientBundle != null) {
+            patientID = patientBundle.getString(ApplicationConstants.BundleKeys.PATIENT_ID_BUNDLE).toString()
+        }
+        val countries = listOf(*resources.getStringArray(R.array.countries_array))
+        var applicationInfo: ApplicationInfo? = null
```
Comment (SalahoAmro) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientActivity.kt:65]: Would be more readable if separated between irrelevant code lines, like if we add an empty line after and before line 64

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientActivity.kt [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+        }
+        val bundle = applicationInfo!!.metaData
+        val googleMapToken = bundle.getString("com.google.android.geo.API_KEY")
+        if (!Places.isInitialized()) {
+            Places.initialize(applicationContext, googleMapToken!!)
+        }
+        val placesClient = Places.createClient(this)
+
+        // Create the mPresenter
+        Log.i("ADDEDIT", "$patientID and $addEditPatientFragment")
```
Comment (SalahoAmro) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientActivity.kt:79]: Would be better if we remove this unnecessary comment, too
Also, remove this log.

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientFragment.java [hunk 4]
```diff
@@ -382,15 +381,15 @@ public void setProgressBarVisibility(boolean visibility) {
     }
```
Comment (SalahoAmro) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientFragment.java:382]: About this file, I read your problem in the comments of your JIRA ticket, that AddEditPatientFragmentPermissionsDispatcher is removed and marked red when you convert this file to a Kotlin. In fact, it is not removed, you can just: Build > Rebuild project.
 If not working try: File > Invalid Caches/Restart.

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/SimilarPatientsRecyclerViewAdapter.kt [hunk 5]
```diff
@@ -0,0 +1,135 @@
+/*  * The contents of this file are subject to the OpenMRS Public License  * Version 1.0 (the "License"); you may not use this file except in  * compliance with the License. You may obtain a copy of the License at  * http://license.openmrs.org  *  * Software distributed under the License is distributed on an "AS IS"  * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the  * License for the specific language governing rights and limitations  * under the License.  *  * Copyright (C) OpenMRS, LLC.  All Rights Reserved.  */ package org.openmrs.mobile.activities.addeditpatient
```
Comment (SalahoAmro) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/SimilarPatientsRecyclerViewAdapter.kt:1]: Please move down the package statement one line.

**Comment by SalahoAmro** [comment]
Heads up! :+1:

**Comment by Rohit-2602** [comment]
@SalahoAmro I try rebuilding the project and file -> Invalidate Caches/ Restart -> Invalidate and restart
but the error is still there (unresolved reference).
Sorry, i messed up in commiting file, I forgot about moving the packing line down in RecyclerView.

**Comment by SalahoAmro** [comment]
@Rohit-2602 What I did is using "convert Java to Kotlin" and found that problem, then undid these changes, rebuilt the project, and converted again to Kotlin, which gave no problems.
Anyway, I don't think it is a problem; I face this issue with Android Studio sometimes that no reference to some classes although it's working. So, as long as you can build the project and run the app, everything is fine.

**Comment by Rohit-2602** [comment]
I converted java to kotlin then i rebuild, doesn't work , i also try (Invalidate Caches/ Restart) that you suggest, again rebuild project.
Try to run the app again still doesn't work.

**Review by Akshatji800 — COMMENTED** [review]
@Rohit-2602 just a suggestion according to the contribution guidelines one needs to squash the commits into a single commit if possible as I see 4 commits here. You can have a look through the "Squashing your commits" section in [this guide ](https://github.com/openmrs/openmrs-contrib-android-client/blob/master/CONTRIBUTING.md)to get an idea about how to squash commits if u don't have the idea. If you have not squashed the commits purposely due to some reason then I apologize for giving a useless suggestion but then if you have done that purposely then why have you checked the "have a single commit" checkbox?

**Comment by Rohit-2602** [comment]
@Akshatji800 Thanks for the suggestion, i will keep that in mind, and try that my PR contain only single commit.

**Comment by SalahoAmro** [comment]
> @Rohit-2602 just a suggestion according to the contribution guidelines one needs to squash the commits into a single commit if possible as I see 4 commits here. You can have a look through the "Squashing your commits" section in [this guide ](https://github.com/openmrs/openmrs-contrib-android-client/blob/master/CONTRIBUTING.md)to get an idea about how to squash commits if u don't have the idea. If you have not squashed the commits purposely due to some reason then I apologize for giving a useless suggestion but then if you have done that purposely then why have you checked the "have a single commit" checkbox?

I read from PR guidelines that it's ok to have multiple commits _after reviews_, and he had only 1 at first submission, which is what he needed at first, IMO.
I appreciate your note, though!

**Comment by LuGO0** [comment]
@Rohit-2602  Closing this for now. please open the PR again after fixing the merge conflicts. thanks

**Pull request closed without merging**